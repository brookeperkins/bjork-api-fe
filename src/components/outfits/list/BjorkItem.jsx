import React from 'react';
import PropTypes from 'prop-types';

const BjorkItem = ({ imgUrl, year }) => (
  <figure>
    <img src={imgUrl} alt={imgUrl} />
    <h3>{year}</h3>
  </figure>
);

BjorkItem.propTypes = {
  imgUrl: PropTypes.string.isRequired,
  year: PropTypes.number.isRequired
};

export default BjorkItem;
