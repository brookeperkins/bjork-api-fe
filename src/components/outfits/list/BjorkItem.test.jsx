import React from 'react';
import { render, cleanup } from '@testing-library/react';
import BjorkItem from './BjorkItem';

describe('<BjorkItem />', () => {
  afterEach(() => cleanup());

  it('renders a single list item', () => {
    const { asFragment } = render(<BjorkItem
      imgUrl="bjork.jpg"
      year={1999}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
