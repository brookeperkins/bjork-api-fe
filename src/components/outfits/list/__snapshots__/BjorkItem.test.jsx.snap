// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BjorkItem /> renders a single list item 1`] = `
<DocumentFragment>
  <figure>
    <img
      alt="bjork.jpg"
      src="bjork.jpg"
    />
    <h3>
      1999
    </h3>
  </figure>
</DocumentFragment>
`;

exports[`BjorkItem component renders single character 1`] = `
<DocumentFragement>
  <figure>
    <img
    alt="bjork.jpg"
    src="bjork.jpg"
    />
    <figcaption>
      1997
    </figcaption>
  </figure>
</DocumentFragement>
`;
